api/server/routes.go

```diff
@@ -72,7 +72,7 @@ func (s *Server) registerQueueRoutes(r *mux.Router) {
 	r.Handle(queueIndexPath, s.jamSessionRequired(http.HandlerFunc(s.getQueue))).Methods("GET")
 	r.Handle(queueCollectionPath, s.jamSessionRequired(s.hostRequired(http.HandlerFunc(s.addCollection)))).Methods("PUT")
 	r.Handle(queueVotePath, s.jamSessionRequired(http.HandlerFunc(s.vote))).Methods("PUT")
-	r.Handle(queueDeletePath, s.jamSessionRequired(s.hostRequired(http.HandlerFunc(s.deleteSong)))).Methods("DELETE")
+	r.Handle(queueDeletePath, s.jamSessionRequired(s.hostRequired(http.HandlerFunc(s.deleteSong)))).Methods("PUT")
```

Comment:
Why is this necessary? Since we are not conforming to REST principles anyways, this change is redundant.

Reply:
We don't, but the Angular http service for example. It does not support delete requests with a body. Also, the delete method was a misfit. We should stick to get and put requests only.

Maybe one day well change to /queue/jamlabel/spotifyid. But atm I don't see an advantage over this method

Reply:
The Angular http client actually does support using bodies on DELETE requests:

https://stackoverflow.com/a/40857437

I would prefer not to change this here, since I would like the API to be at least some kind of stable right now. I would accept the change here, if you are willing to fix the new API call in jamfactory-web as well as in the secret thing.

I don't want to blindly block the change but want you to know the impact it would have. So if you want to fix all places where this change would have an impact, go ahead and do so.

Reply:
Reverted the change for now. On the next occasion we should clean up the method types.

Reply:
And why is the flutter app such a big secret?

Reply:
Because the repository is still private